sql/parser/eval.go

```diff
@@ -134,15 +137,23 @@ var binOps = map[binArgs]binOp{
 		},
 	},
 	binArgs{Plus, dateType, intervalType}: {
-		returnType: DummyTimestamp,
+		returnType: DummyDate,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			days, err := inDays(right.(DInterval).Duration)
+			if err != nil {
+				return DummyDate, err
+			}
+			return DDate(left.(DDate) + days), nil
```

Comment:
I think that remains to be seen. It's probably true that no other system is as strict as we are, but how much does that matter? Most ORMs are pretty conservative in the SQL they generate and would be compatible with a strict server-side typesystem. The place where strictness gets annoying is the interactive console, which is has such different requirements that I wonder if we should offer a special non-strict mode for it.